engine/src/host_imports/ivalues_lifting.rs

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2020 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use super::WType;
+use super::WValue;
+use super::HostImportError;
+use crate::IValue;
+use crate::IType;
+use super::Result;
+
+use wasmer_core::memory::ptr::{Array, WasmPtr};
+use wasmer_core::vm::Ctx;
+use wasmer_wit::types::RecordType;
+use wasmer_wit::vec1::Vec1;
+
+use std::collections::HashMap;
+use std::rc::Rc;
+
```

Comment:
It would be nice to have a small comment on the meaning of the whole file, maybe on what "lifting" means in this context.